muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/actions/workspaces/index.ts

```diff
@@ -762,49 +857,232 @@ let updateWorkspace:UpdateWorkspaceTriggerType = function updateWorkspace(data){
         }
       });
 
-      if (!(err instanceof MApiError)){
+      if (!(err instanceof MApiError)) {
+        throw err;
+      }
+      dispatch(displayNotification(getState().i18n.text.get('plugin.workspace.management.notification.save.error'), 'error'));
+
+      data.fail && data.fail();
+    }
+  }
+}
+
+let updateOrganizationWorkspace: UpdateWorkspaceTriggerType = function updateOrganizationWorkspace(data) {
+  return async (dispatch: (arg: AnyActionType) => any, getState: () => StateType) => {
+    try {
+
+      let originalWorkspace = data.workspace;
+
+      // Take off data that'll cramp the update
+      delete originalWorkspace["staffMemberSelect"];
+      delete originalWorkspace["studentsSelect"];
+
+      if (data.update) {
+        await promisify(mApi().workspace.workspaces.update(data.workspace.id,
+          Object.assign(data.workspace, data.update)
+        ), 'callback')().then(
+          data.success && data.success("WORKSPACE-UPDATE")
+        );
+      }
+
+      if (data.addStudents.length > 0) {
+        let groupIdentifiers: number[] = [];
+        let studentIdentifiers: string[] = [];
+
+        data.addStudents.map(student => {
+          if (student.type === "student-group") {
+            groupIdentifiers.push(student.id as number);
+          }
+          if (student.type === "student") {
+            studentIdentifiers.push(student.id as string);
+          }
+        });
+
+        await promisify(mApi().organizationmanagement.workspaces.students
+          .create(data.workspace.id, {
+            studentIdentifiers: studentIdentifiers,
+            studentGroupIds: groupIdentifiers
+          }
+          ), 'callback')().then(
+            data.success && data.success("ADD-STUDENTS")
+          );
+      }
+
+      if (data.addTeachers.length > 0) {
+        let staffMemberIdentifiers = data.addTeachers.map(teacher => teacher.id);
+
+        await promisify(mApi().organizationmanagement.workspaces.staff
+          .create(data.workspace.id, {
+            staffMemberIdentifiers: staffMemberIdentifiers
+          }
+          ), 'callback')().then(
+            data.success && data.success("ADD-TEACHERS")
+          );
+      }
+
+      if (data.removeStudents.length > 0) {
```

Comment:
All this added code is odd and doesn't seem to do anything, why is data.success called so many times anyway?... do we have a progressbar of the sorts?... then it should be data.progress... not success

Reply:
Yes, you are correct. Fixing this.